ftbfs with gcc 10

It fails to build unless I pass CPPFLAGS="-fcommon" (gcc>=10 defaults to -fno-common):
```
==> Starting build()...
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c main.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c main_menu.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c serial.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c options.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c sensors.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c trouble_code_reader.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c custom_gui.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c error_handlers.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c about.c
gcc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -c reset.c
gcc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Werror -DTERMIOS -o scantool main.o main_menu.o serial.o options.o sensors.o trouble_code_reader.o custom_gui.o error_handlers.o about.o reset.o -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -L/usr/lib -lalleg -lm
/usr/bin/ld: main_menu.o:(.bss+0x58): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x0): multiple definition of `comport'; main.o:(.bss+0x0): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x18): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: main_menu.o:(.bss+0xc): multiple definition of `serial_timer_running'; main.o:(.bss+0xc): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x10): multiple definition of `serial_time_out'; main.o:(.bss+0x10): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x20): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x38): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x40): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x48): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x50): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: main_menu.o:(.bss+0x54): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
/usr/bin/ld: serial.o:(.bss+0x10): multiple definition of `serial_time_out'; main.o:(.bss+0x10): first defined here
/usr/bin/ld: serial.o:(.bss+0xc): multiple definition of `serial_timer_running'; main.o:(.bss+0xc): first defined here
/usr/bin/ld: serial.o:(.bss+0x0): multiple definition of `comport'; main.o:(.bss+0x0): first defined here
/usr/bin/ld: serial.o:(.bss+0x18): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: serial.o:(.bss+0x20): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: serial.o:(.bss+0x38): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: serial.o:(.bss+0x40): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: serial.o:(.bss+0x48): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: serial.o:(.bss+0x50): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: serial.o:(.bss+0x54): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
/usr/bin/ld: serial.o:(.bss+0x58): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: options.o:(.bss+0x154): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
/usr/bin/ld: options.o:(.bss+0x150): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: options.o:(.bss+0x100): multiple definition of `comport'; main.o:(.bss+0x0): first defined here
/usr/bin/ld: options.o:(.bss+0x148): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: options.o:(.bss+0x118): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: options.o:(.bss+0x0): multiple definition of `temp_error_buf'; main.o:(.bss+0x20): first defined here
/usr/bin/ld: options.o:(.bss+0x10c): multiple definition of `serial_timer_running'; main.o:(.bss+0xc): first defined here
/usr/bin/ld: options.o:(.bss+0x110): multiple definition of `serial_time_out'; main.o:(.bss+0x10): first defined here
/usr/bin/ld: options.o:(.bss+0x120): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: options.o:(.bss+0x138): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: options.o:(.bss+0x140): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: options.o:(.bss+0x158): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: sensors.o:(.bss+0x154): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
/usr/bin/ld: sensors.o:(.bss+0x0): multiple definition of `temp_error_buf'; main.o:(.bss+0x20): first defined here
/usr/bin/ld: sensors.o:(.bss+0x100): multiple definition of `comport'; main.o:(.bss+0x0): first defined here
/usr/bin/ld: sensors.o:(.bss+0x158): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: sensors.o:(.bss+0x110): multiple definition of `serial_time_out'; main.o:(.bss+0x10): first defined here
/usr/bin/ld: sensors.o:(.bss+0x10c): multiple definition of `serial_timer_running'; main.o:(.bss+0xc): first defined here
/usr/bin/ld: sensors.o:(.bss+0x118): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: sensors.o:(.bss+0x120): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: sensors.o:(.bss+0x138): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: sensors.o:(.bss+0x140): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: sensors.o:(.bss+0x148): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: sensors.o:(.bss+0x150): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x118): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x138): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x100): multiple definition of `comport'; main.o:(.bss+0x0): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x0): multiple definition of `temp_error_buf'; main.o:(.bss+0x20): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x110): multiple definition of `serial_time_out'; main.o:(.bss+0x10): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x10c): multiple definition of `serial_timer_running'; main.o:(.bss+0xc): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x120): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x140): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x148): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x150): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x154): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
/usr/bin/ld: trouble_code_reader.o:(.bss+0x158): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: custom_gui.o:(.bss+0x0): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: custom_gui.o:(.bss+0x10): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: custom_gui.o:(.bss+0x28): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: custom_gui.o:(.bss+0x30): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: custom_gui.o:(.bss+0x38): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: custom_gui.o:(.bss+0x40): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: custom_gui.o:(.bss+0x44): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
/usr/bin/ld: custom_gui.o:(.bss+0x48): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x100): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x0): multiple definition of `temp_error_buf'; main.o:(.bss+0x20): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x110): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x128): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x130): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x138): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x140): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x144): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
/usr/bin/ld: error_handlers.o:(.bss+0x148): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: about.o:(.bss+0x18): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: about.o:(.bss+0x0): multiple definition of `comport'; main.o:(.bss+0x0): first defined here
/usr/bin/ld: about.o:(.bss+0x10): multiple definition of `serial_time_out'; main.o:(.bss+0x10): first defined here
/usr/bin/ld: about.o:(.bss+0xc): multiple definition of `serial_timer_running'; main.o:(.bss+0xc): first defined here
/usr/bin/ld: about.o:(.bss+0x20): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: about.o:(.bss+0x38): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: about.o:(.bss+0x40): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: about.o:(.bss+0x48): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: about.o:(.bss+0x50): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: about.o:(.bss+0x54): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
/usr/bin/ld: about.o:(.bss+0x58): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: reset.o:(.bss+0xc): multiple definition of `serial_timer_running'; main.o:(.bss+0xc): first defined here
/usr/bin/ld: reset.o:(.bss+0x10): multiple definition of `serial_time_out'; main.o:(.bss+0x10): first defined here
/usr/bin/ld: reset.o:(.bss+0x58): multiple definition of `is_not_genuine_scan_tool'; main.o:(.bss+0x168): first defined here
/usr/bin/ld: reset.o:(.bss+0x0): multiple definition of `comport'; main.o:(.bss+0x0): first defined here
/usr/bin/ld: reset.o:(.bss+0x18): multiple definition of `datafile'; main.o:(.bss+0x120): first defined here
/usr/bin/ld: reset.o:(.bss+0x20): multiple definition of `log_file_name'; main.o:(.bss+0x130): first defined here
/usr/bin/ld: reset.o:(.bss+0x38): multiple definition of `code_defs_file_name'; main.o:(.bss+0x148): first defined here
/usr/bin/ld: reset.o:(.bss+0x40): multiple definition of `data_file_name'; main.o:(.bss+0x150): first defined here
/usr/bin/ld: reset.o:(.bss+0x48): multiple definition of `options_file_name'; main.o:(.bss+0x158): first defined here
/usr/bin/ld: reset.o:(.bss+0x50): multiple definition of `display_mode'; main.o:(.bss+0x160): first defined here
/usr/bin/ld: reset.o:(.bss+0x54): multiple definition of `system_of_measurements'; main.o:(.bss+0x164): first defined here
collect2: error: ld returned 1 exit status
make: *** [makefile:56: scantool] Error 1
```

--- a/error_handlers.c
+++ b/error_handlers.c
@@ -2,6 +2,7 @@
 #include "globals.h"
 #include "error_handlers.h"
 
+char temp_error_buf[256];
 
 void fatal_error(char *msg)
 {

--- a/error_handlers.h
+++ b/error_handlers.h
@@ -1,7 +1,7 @@
 #ifndef ERROR_HANDLERS_H
 #define ERROR_HANDLERS_H
 
-char temp_error_buf[256];
+extern char temp_error_buf[256];
 
 void fatal_error(char *msg);
 

--- a/globals.h
+++ b/globals.h
@@ -36,9 +36,6 @@
 #define C_GRAY          50
 #define C_LIGHT_GRAY    55
 
-//
-int is_not_genuine_scan_tool;
-
 // Options
 int system_of_measurements;
 int display_mode;

--- a/globals.h
+++ b/globals.h
@@ -37,17 +37,12 @@
 #define C_LIGHT_GRAY    55
 
 // Options
-int system_of_measurements;
-int display_mode;
+extern int system_of_measurements;
+extern int display_mode;
 
 // File names
-char *options_file_name;
-char *data_file_name;
-char *code_defs_file_name;
-char log_file_name[20];
-#ifdef LOG_COMMS
-char comm_log_file_name[20];
-#endif
+extern char *options_file_name;
+extern char *code_defs_file_name;
 
 void write_log(const char *log_string);
 
@@ -55,6 +50,6 @@ void write_log(const char *log_string);
 void write_comm_log(const char *marker, const char *data);
 #endif
 
-DATAFILE *datafile;
+extern DATAFILE *datafile;
 
 #endif

--- a/main.c
+++ b/main.c
@@ -48,8 +48,6 @@ static void init()
 {
    char temp_buf[256];
 
-   is_not_genuine_scan_tool = FALSE;
-   
    /* initialize some varaibles with default values */
    if (asprintf(&options_file_name, "%s/.scantoolrc", getenv("HOME")) < 0) {
       perror("asprintf");

--- a/main.c
+++ b/main.c
@@ -16,6 +16,13 @@ END_COLOR_DEPTH_LIST
 
 #define WINDOW_TITLE   "ScanTool.net " SCANTOOL_VERSION_EX_STR
 
+int display_mode;
+char *code_defs_file_name;
+char log_file_name[20];
+#ifdef LOG_COMMS
+char comm_log_file_name[20];
+#endif
+DATAFILE *datafile;
 
 void write_log(const char *log_string)
 {
@@ -47,6 +54,7 @@ void write_comm_log(const char *marker, const char *data)
 static void init()
 {
    char temp_buf[256];
+   char *data_file_name;
 
    /* initialize some varaibles with default values */
    if (asprintf(&options_file_name, "%s/.scantoolrc", getenv("HOME")) < 0) {

--- a/main_menu.c
+++ b/main_menu.c
@@ -298,9 +298,5 @@ int button_desc_proc(int msg, DIALOG *d, int c)
 
 int genuine_proc(int msg, DIALOG *d, int c)
 {
-   if (msg == MSG_IDLE)
-      if (is_not_genuine_scan_tool)
-         return D_CLOSE;
-   
    return D_O_K;
 }

--- a/options.c
+++ b/options.c
@@ -30,6 +30,9 @@
    #define DEFAULT_BAUD_RATE               BAUD_RATE_115200
 #endif
 
+int system_of_measurements;
+char *options_file_name;
+
 typedef struct
 {
    int option_value;

--- a/reset.c
+++ b/reset.c
@@ -283,7 +283,6 @@ int Reset_get_reply_to_detect_protocol(char *response)
 /* We don't care if it's a clone; let it run anyway. */
 int Reset_handle_clone()
 {
-   is_not_genuine_scan_tool = FALSE;
    return RESET_START_ECU_TIMER;
 }
 

--- a/sensors.c
+++ b/sensors.c
@@ -1493,9 +1493,5 @@ void clr_time_formula(int data, char *buf)
 
 int genuine_proc(int msg, DIALOG *d, int c)
 {
-   if (msg == MSG_IDLE)
-      if (is_not_genuine_scan_tool == TRUE)
-         return D_CLOSE;
-   
    return D_O_K;
 }

--- a/serial.c
+++ b/serial.c
@@ -24,7 +24,9 @@
 #define TX_TIMEOUT_MULTIPLIER    0
 #define TX_TIMEOUT_CONSTANT      1000
 
-
+volatile int serial_time_out;
+volatile int serial_timer_running;
+struct COMPORT comport;
 
 //timer interrupt handler for sensor data
 static void serial_time_out_handler()

--- a/serial.h
+++ b/serial.h
@@ -77,10 +77,10 @@ const char *get_protocol_string(int interface_type, int protocol_id);
 int display_error_message(int error, int retry);
 
 // variables
-volatile int serial_time_out;
-volatile int serial_timer_running;
+extern volatile int serial_time_out;
+extern volatile int serial_timer_running;
 
-struct COMPORT {
+extern struct COMPORT {
    int number;
    int baud_rate;
    int status;    // READY, NOT_OPEN, USER_IGNORED

